### PR TITLE
depends: Fix cross build for qt5.7

### LINKED ISSUE #9585: Can't build qt 5.7.1 for windows
### Describe the issue
After the qt 5.7.1 update i haven't been able to build gui for windows. Im using ubuntu 14.04 vm and I had no issues building qt 5.6.1.
#### If so, please list the steps to reproduce below:
1. I follow the windows build notes.

```
checking for main in -lQt5Core   ... yes
checking for main in -lQt5Gui    ... yes
checking for main in -lQt5Network... yes
checking for main in -lQt5Widgets... yes
checking for main in -lQt5Test... yes
checking QTest usability... yes
checking QTest presence... yes
checking for QTest... yes
checking for main in -lQt5DBus... yes
checking QtDBus usability... yes
checking QtDBus presence... yes
checking for QtDBus... yes
checking for static Qt... yes
checking for Qt >= 5.6... yes
checking for main in -lQt5PlatformSupport... yes
checking for Qt < 5.4... no
checking for static Qt plugins: -lqwindows... no
configure: WARNING: Could not resolve: -lqwindows; bitcoin-qt frontend will not be built
checking whether to build Bitcoin Core GUI... no (Qt5)
```
Is there some extra package you now have to install? I have installed these: libqt5gui5 libqt5core5a libqt5dbus5 qttools5-dev qttools5-dev-tools libprotobuf-dev protobuf-compiler
- OS: Windows 10 x64

### PR DESCRIPTION
**Pull request by theuni**
Fixes #9585.

The qt build is a mess, but this gets it building for 5.7. Going forward, we need to split it up into smaller parts, where the native tools (qmake/rcc/uic/etc.) are separate from the target libs. That will let us eliminate all of the cross-build hacks.

Includes a temp commit to have Travis build for Windows.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Concept ACK.

The travis log mentions, that qt was detected:

```
Options used to compile and link:
  with wallet   = yes
  with gui / qt = yes
    qt version  = 5
    with qr     = yes
  with zmq      = yes
  with test     = yes
  with bench    = yes
  with upnp     = yes
  debug enabled = no

  target os     = windows
  build os      = 

  CC            = /home/travis/build/bitcoin/bitcoin/depends/x86_64-w64-mingw32/share/../native/bin/ccache x86_64-w64-mingw32-gcc
  CFLAGS        = -pipe -O2 
  CPPFLAGS      = -I/home/travis/build/bitcoin/bitcoin/depends/x86_64-w64-mingw32/share/../include/  -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS -D_MT -DWIN32 -D_WINDOWS -DBOOST_THREAD_USE_LIB -D_FILE_OFFSET_BITS=64
  CXX           = /home/travis/build/bitcoin/bitcoin/depends/x86_64-w64-mingw32/share/../native/bin/ccache x86_64-w64-mingw32-g++ -std=c++11
  CXXFLAGS      = -pipe -O2  -fvisibility=hidden
  LDFLAGS       = -L/home/travis/build/bitcoin/bitcoin/depends/x86_64-w64-mingw32/share/../lib 

```

So I guess you are fine to drop the last commit?

**Comment by theuni** [comment]
Done

File: build-aux/m4/bitcoin_qt.m4 [hunk 1]
```diff
@@ -473,8 +473,8 @@ AC_DEFUN([_BITCOIN_QT_FIND_LIBS_WITHOUT_PKGCONFIG],[
   ])
 
   BITCOIN_QT_CHECK(AC_CHECK_LIB([z] ,[main],,AC_MSG_WARN([zlib not found. Assuming qt has it built-in])))
-  BITCOIN_QT_CHECK(AC_CHECK_LIB([png] ,[main],,AC_MSG_WARN([libpng not found. Assuming qt has it built-in])))
-  BITCOIN_QT_CHECK(AC_CHECK_LIB([jpeg] ,[main],,AC_MSG_WARN([libjpeg not found. Assuming qt has it built-in])))
+  BITCOIN_QT_CHECK(AC_SEARCH_LIBS([png_error] ,[qtpng png],,AC_MSG_WARN([libpng not found. Assuming qt has it built-in])))
+  BITCOIN_QT_CHECK(AC_SEARCH_LIBS([jpeg_create_decompress] ,[qtjpeg jpeg],,AC_MSG_WARN([libjpeg not found. Assuming qt has it built-in])))
```
Comment (MarcoFalke) [inline build-aux/m4/bitcoin_qt.m4]: We are not using jpeg, so this seems unused.

**Comment by laanwj** [comment]
Tested ACK b5f374f.
I built with gitian and can see that the -qt executables are in the resulting .zip. I have however not tried to test them on windows, but the build works.

**Comment by fanquake** [comment]
I can't test this at the moment, but the changes look ok. Splitting up the
Qt build has been on the cards for a while now.

utACK.

**Comment by achow101** [comment]
Post-merge ACK

Built with gitian and tested on Windows 10

**Pull request merged**